doc/api/process.md

```diff
@@ -706,26 +706,69 @@ Example:
 
 ## process.exit([code])
 
-Ends the process with the specified `code`.  If omitted, exit uses the
-'success' code `0`.
+* `code` {Integer} The exit code. Defaults to `0`.
+
+The `process.exit()` methods instructs Node.js to terminate the process as
+quickly as possible with the specified exit `code`. If the `code` is omitted, 
+exit uses either the 'success' code `0` or the value of `process.exitCode` if
+specified.
 
 To exit with a 'failure' code:
 
 ```js
 process.exit(1);
 ```
 
-The shell that executed Node.js should see the exit code as 1.
+The shell that executed Node.js should see the exit code as `1`.
+
+It is important to note that calling `process.exit()` will force the process to
+exit as quickly as possible *even if there are still asynchronous operations 
+pending* in the event loop, *including* i/o operations to `process.stdout` and
+`process.stderr`.
```

Comment:
Yep. Tried to capture some of those nuances in my first edit but gave up because it ended up being too confusing. Open to wording that works better.